uhi/typing/plottable.py

```diff
@@ -46,10 +59,20 @@ def __eq__(self, other: Any) -> bool:
         """
 
 
+PlottableAxisContinuous = PlottableAxisGeneric[Tuple[float, float]]
+PlottableAxisInt = PlottableAxisGeneric[int]
+PlottableAxisStr = PlottableAxisGeneric[str]
+
+PlottableAxis = Union[PlottableAxisContinuous, PlottableAxisInt, PlottableAxisStr]
+
+
 @runtime_checkable
 class PlottableHistogram(Protocol):
     axes: Sequence[PlottableAxis]
 
+    weighted: bool
+    interpretation: Literal["count", "mean", "median"]
```

Comment:
Better use an enum?

Reply:
Do we have a histogram that computes the median already?

Reply:
This can't be an enum because it's a protocol; where would you put it? If you are plotting an Uproot histogram with mplhep, you can't depend an an enum from boost-histogram, for example. And "median" isn't supported by boost-histogram (yet?), but Jim wanted it for future proofing since it's a common type that is used in some other areas.

Reply:
@jpivarski Where is median a common type?

Reply:
I think the protocol right now should only contain the least common denominator of what is needed now. More literals can be added to `interpretation` at any later time.

Reply:
> This can't be an enum because it's a protocol; where would you put it?

The protocol requires interface to be present, it can also require that an enum with certain field exists.

Reply:
Every library would have their own enum, just like every library has their own classes that implement the protocol. Strings are bad.

Reply:
Python uses string comparisons everywhere, it's harder to find examples of enums than of strings. For example, matplotlib's `plt.plot` has strings everywhere, such as`'default', 'steps', 'steps-pre', 'steps-mid', 'steps-post'` for drawstyle, etc. So using a string would be a reasonable Pythonic thing to do.

> it can also require that an enum with certain field exists.

No, it cannot. For example:

```python
class Enum1(enum.Enum):
    one = 1
class Enum2(enum.Enum):
    one = 1

Enum1.one == Enum2.one
# FALSE  - these are unique
```

The point of an enum is that is unique, you can't make one up without using it. Since the whole point of a Protocol is to talk between libraries without introducing a dependence, this won't work! The plotting library's enum would not match the source library's enum.

If you are comparing the string-name of the enum fields, besides misusing enums in a way they were not designed, that's also much less convenient than just using a string in the first place!

Also, while performance doesn't matter - this is not even a problem either. In Python, short strings are interned, and when you compare, it just compares the pointer. Both of these are short enough:

```python
>>> id("mean")
4482494000
>>> id("mean")
4482494000
>>> id("count")
4479637616
>>> id("count")
4479637616
```

Now, when defining this in a library, I _think_ a valid implementation would be:

```python
class StorageType(str, enum.Enum):
    count = "count"
    mean = "mean"
```

But I _think_ this is a valid implementation of this, and would be up to the library implementing it. (Need to verify it would pass static and dynamic isinstance correctly, but I think it should - might be a bug if it doesn't.)

Reply:
PS: If it's not clear from context, a str mixin enum compares with the value, that is, `StorageType.count == "count"` is True.

Reply:
Ugh, ran into an issue if this is implemented as a string enum by a client library. `Enum.count` is not valid for a [StrEnum](https://docs.python.org/3/library/enum.html#others) - as `.count` is defined on a string, so Enum.count.count doesn't make any sense - the first count should be an enum instance and the second count would be a string method, but it's actually Enum.count again. How about `sum` and `mean` instead? Or all-caps?

Reply:
The reason I was against "sum" is because an accumulator might be written that sums all `x` values given by `fill(x)` or `y` in `fill(x, y)` (times any weights given as an optional last parameter, not shown here). That's not what this does: it sums `1` (possibly times weight) for every `fill(x)`, with the `x` value only used to determine which bin it should go in. So Histogrammar distinguished between (possibly weighted) "count" and "sum", though "count" is by far more common.

One option would be to pluralize all of the choices: "counts," maybe "sums," definitely "means" (for profiles). Or maybe use more traditional language. Only "counts" gives you a true histogram, so "histogram" vs "profile" vs other names. (What, in that scheme, would the name for medians be?)

Or maybe capitalization if the equality of these strings is taken to be case-insensitive, with the capitalized form as canonical. Then the enums would appear in capitalized form (not unexpected, since they're constant symbols), but strings are allowed to be lowercase (which feels more natural—or maybe just more modern).